x-pack/plugins/enterprise_search/server/lib/ml/get_ml_model_deployment_status.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  MlGetTrainedModelsStatsRequest,
+  MlGetTrainedModelsRequest,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { MlTrainedModels } from '@kbn/ml-plugin/server';
+
+import {
+  MlModelDeploymentStatus,
+  MlModelDeploymentState,
+  MlTrainedModelConfigWithDefined,
+} from '../../../common/types/ml';
+
+import { isNotFoundExceptionError } from './ml_model_deployment_common';
+
+export const getMlModelDeploymentStatus = async (
+  modelName: string,
+  trainedModelsProvider: MlTrainedModels | undefined
+): Promise<MlModelDeploymentStatus> => {
+  if (!trainedModelsProvider) {
+    throw new Error('Machine Learning is not enabled');
```

Comment:
@demjened I think this is ok since we don't fetch model status if ml is disabled, but we should ensure thats the case.